### ISSUE TITLE
hdf5: HDF5_z_LIBRARY (zlib) dependency not found during build depending on hdf5

### ISSUE THREAD

**Issue by knedlsepp** [NixOS/nixpkgs#17099]
## Issue description
On building ([my proposed changes of](https://github.com/NixOS/nixpkgs/pull/17076/commits/c9cbbc38936679aab09b82dcaf9a42611cc7fda3)) the `armadillo` package, which depends on `hdf5`, cmake's builtin FindHDF5 kicks off. The cmake process fails with the error:

```
CMake Error: The following variables are used in this project, 
but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
HDF5_z_LIBRARY
```

### Steps to reproduce

- Check out: https://github.com/NixOS/nixpkgs/pull/17076/commits/c9cbbc38936679aab09b82dcaf9a42611cc7fda3
- Build the `armadillo` derivation. 

### Proposed solution
I am not sure if this is the right way to go, but the problem can be fixed by specifying `zlib` as a propagatedBuildInput for hdf5:
https://github.com/NixOS/nixpkgs/pull/17076/commits/e9573cb8f4f5b0faf806b77f53fd872b389dd486

## Technical details

* System: NixOS: 16.03.657.0c12bab (Emu)
* Nix version: nix-env (Nix) 1.11.2
* Nixpkgs version: Current master

**Comment by FRidh** [comment]
Just testing some other changes as well and then I'll push the change.

**Comment by FRidh** [comment]
fixed in 46378eb128b3e64152a75bdb36eef8399e87558d

**Comment by knedlsepp** [comment]
Thanks!
